### PR TITLE
Added nightly toolchain step to setup docs

### PR DESCRIPTION
**Pull request by jackbackes**
Minor change to address #201 by adding a step to developer setup to install the rust nightly toolchain.

### CODE REVIEW

File: docs/docs/develop.md [hunk 1]
```diff
@@ -11,6 +11,13 @@ Install various rust related tools -
 * Rust Compiler - <http://rustup.rs>
 * Cargo tools - clippy, rustfmt is very helpful for formating and fixing warnings.
 
+Install the "nightly" toolchain -
+
+```shell
+rustup update nightly
+rustup override set nightly
```
Comment (diptanu) [inline docs/docs/develop.md:18]: Do we need this? We don't depend on anything from the nightly other than formatting stuff. 

I have a make target in the Makefile which does cargo @nightly fmt or something like that which triggers the nightly toolchain only formatting.

**Review by diptanu — CHANGES_REQUESTED** [review]
One small comment.

File: docs/docs/develop.md (hunk 1 above)
Comment (jackbackes) [reply docs/docs/develop.md:18]: I ran into this issue when doing a fresh build for development setup. There were unsupported dependencies that required the nightly toolchain. IIRC it had to do with sea-orm.

I'm rebuilding it fresh so I'll see if I run into the issue again.

Comment (jackbackes) [reply docs/docs/develop.md:18]: I'm going to close this as "can't reproduce." I reinstalled and ran Indexify in a fresh environment and it worked fine. Maybe I had an out of date version of rustc. That will teach me to not use the dev container. ¯\_(ツ)_/¯

**Comment by jackbackes** [comment]
Can't reproduce.

**Pull request closed without merging**